shopfloor/tests/test_checkout_select.py

```diff
@@ -52,7 +52,7 @@ def _test_error(self, picking, msg):
 
     def test_select_error_not_found(self):
         picking = self._create_picking()
-        picking.unlink()
+        picking.sudo().unlink()
         self._test_error(picking, "This transfer does not exist anymore.")
```

Comment:
I'm not sure to understand how the `_test_error` is working as expected when it performs the call `_picking_summary_data` which "jsonify" the picking, I was thinking it should crash if the recordset data are read while the it is actually removed.
But if it works, I won't look further... :)

Reply:
It's actually what it tests: `_picking_summary_data` should not return the jsonified picking but return to selection of picking (state: `manual_selection`) with an error message ("This transfer does not exist anymore.") and the details of pickings that still exist (`self.picking` here, which we didn't delete)